toolkit/tools/imagecustomizer/main.go

```diff
@@ -4,57 +4,77 @@
 package main
 
 import (
-	"fmt"
 	"log"
-	"os"
+	"maps"
 	"strings"
 
+	"github.com/alecthomas/kong"
 	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
-	"github.com/microsoft/azurelinux/toolkit/tools/internal/exe"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/exekong"
 	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/ptrutils"
 	"github.com/microsoft/azurelinux/toolkit/tools/internal/timestamp"
 	"github.com/microsoft/azurelinux/toolkit/tools/pkg/imagecustomizerlib"
-	"gopkg.in/alecthomas/kingpin.v2"
 )
 
-var (
-	app = kingpin.New("imagecustomizer", "Customizes a pre-built Azure Linux image")
-
-	buildDir                 = app.Flag("build-dir", "Directory to run build out of.").Required().String()
-	inputImageFile           = app.Flag("image-file", "Path of the base Azure Linux image which the customization will be applied to.").String()
-	outputImageFile          = app.Flag("output-image-file", "Path to write the customized image to.").String()
-	outputImageFormat        = app.Flag("output-image-format", fmt.Sprintf("Format of output image. Supported: %s.", strings.Join(imagecustomizerapi.SupportedImageFormatTypes(), ", "))).Enum(imagecustomizerapi.SupportedImageFormatTypes()...)
-	configFile               = app.Flag("config-file", "Path of the image customization config file.").Required().String()
-	rpmSources               = app.Flag("rpm-source", "Path to a RPM repo config file or a directory containing RPMs.").Strings()
-	disableBaseImageRpmRepos = app.Flag("disable-base-image-rpm-repos", "Disable the base image's RPM repos as an RPM source").Bool()
-	outputPXEArtifactsDir    = app.Flag("output-pxe-artifacts-dir", "Create a directory with customized image PXE booting artifacts. '--output-image-format' must be set to 'iso'.").String()
-	logFlags                 = exe.SetupLogFlags(app)
-	timestampFile            = app.Flag("timestamp-file", "File that stores timestamps for this program.").String()
-)
+type CustomizeCmd struct {
+	BuildDir                 string   `name:"build-dir" help:"Directory to run build out of." required:""`
+	InputImageFile           string   `name:"image-file" help:"Path of the base Azure Linux image which the customization will be applied to."`
```

Comment:
tip: for all these files you are expecting should exist you can add `type:existingFile` and kong will check that for you.
(https://github.com/alecthomas/kong?tab=readme-ov-file#custom-named-decoders)

Reply:
The `existingFile` parser has some additional behavior that I don't want. For example, on Linux, `~` should be interpreted by the shell not the program.